nautobot/docs/additional-features/jobs.md

```diff
@@ -553,57 +553,90 @@ Provision of user input (`data` values) via the CLI is not supported at this tim
-            settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
+Set up the `"job_logs"` database correctly for testing:
 
-    This code isn't *needed* in Nautobot 1.3.x and later, as the `"job_logs"` database will already be set up correctly for testability, but it also won't hurt anything to include.
+```python
+from django.conf import settings
+
+if "job_logs" in settings.DATABASES:
+    settings.DATABASES["job_logs"] = settings.DATABASES["job_logs"].copy()
+    settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
+  ```
 
-A simple example of a Job test case that's implemented to be backwards-compatible with Nautobot 1.2.x might look like the following:
+Replicate the behavior of `run_job_for_testing` manually so that your test execution most closely resembles the way the celery worker would run the test:
 
 ```python
 import uuid
-
-from django.conf import settings
+from django.contrib.auth import get_user_model
 from django.contrib.contenttypes.models import ContentType
-from django.test import TransactionTestCase
 
+from nautobot.extras.context_managers import web_request_context
 from nautobot.extras.jobs import run_job
-from nautobot.extras.models import Job, JobResult, JobLogEntry
+from nautobot.extras.models import JobResult, Job
+
+def run_job_for_testing(job, data=None, commit=True, username="test-user"):
+    if data is None:
+        data = {}
+    user_model = get_user_model()
+    user, _ = user_model.objects.get_or_create(username=username, is_superuser=True, password="password")
+    job_result = JobResult.objects.create(
+        name=job.class_path,
+        obj_type=ContentType.objects.get_for_model(Job),
+        user=user,
+        job_model=job,
+        job_id=uuid.uuid4(),
+    )
+    with web_request_context(user=user) as request:
+        run_job(data=data, request=request, commit=commit, job_result_pk=job_result.pk)
+    return job_result
+```
 
+Setup the `databases` field on the test class correctly, and re-create the default Statuses on `setUp` in your test classes, because `django.test.TransactionTestCase` truncates them on every `tearDown`:
 
-if "job_logs" in settings.DATABASES:
-    settings.DATABASES["job_logs"] = settings.DATABASES["job_logs"].copy()
-    settings.DATABASES["job_logs"]["TEST"] = {"MIRROR": "default"}
+```python
+from django.apps import apps
+from django.conf import settings
+from django.test import TransactionTestCase
 
+from nautobot.extras.management import populate_status_choices
 
 class MyJobTestCase(TransactionTestCase):
+    # 'job_logs' is a proxy connection to the same (default) database that's used exclusively for Job logging
     if "job_logs" in settings.DATABASES:
         databases = ("default", "job_logs")
 
+    def setUp(self):
+        super().setUp()
+        populate_status_choices(apps, None)
+```
+
+---
+
+A simple example of a Job test case for 1.3.x and forward might look like the following:
```

Comment:
Would it flow better, do you think, to move this example up into the "Nautobot 1.3 and later` section?

Reply:
Yes, that makes a lot of sense, thanks!